custom/enikshay/management/commands/data_dumps_person_case.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.apps.es.case_search import CaseSearchES
+from corehq.apps.es import queries
+
+from custom.enikshay.case_utils import (
+    CASE_TYPE_EPISODE,
+    get_all_occurrence_cases_from_person,
+)
+from custom.enikshay.const import ENROLLED_IN_PRIVATE
+from custom.enikshay.management.commands.base_data_dump import BaseDataDump
+
+DOMAIN = "enikshay"
+
+
+class Command(BaseDataDump):
+    """
+    data dumps for person cases
+    https://docs.google.com/spreadsheets/d/1OPp0oFlizDnIyrn7Eiv11vUp8IBmc73hES7qqT-mKKA/edit#gid=1039030624
+    """
+    @staticmethod
+    def get_custom_value(column_name, case):
+        if column_name == 'Reason for "Remove a Person" / Closure':
+            if case.closed:
+                return "closed"
+            elif case.owner_id == "_invalid_":
+                return "removed"
+            elif case.owner_id == '_archive_':
+                return "archived"
+            else:
+                return "active"
+        elif column_name == 'Latest Episode - Date Closed (If any)':
+            try:
+                last_episode_case = get_last_episode(case)
+                if last_episode_case.closed:
+                    return "closed"
+                else:
+                    return "open"
+            except Exception as e:
+                return str(e)
+
+    def get_case_reference_value(self, case_reference, case, calculation):
+        if case_reference == 'last_episode':
+            try:
+                last_episode_case = get_last_episode(case)
```

Comment:
You can memoize this lookup so it only happens once per case, for instance by having a "context" object that's instantiated on each row for the purpose of caching lookups.

Reply:
Done.  Adding context.